### PR TITLE
perf(compiler-cli): avoid module resolution in cycle analysis

### PR DESCRIPTION
**Pull request by JoostK**
See individual commits.

### CODE REVIEW

File: packages/compiler-cli/src/ngtsc/cycles/src/imports.ts [hunk 1]
```diff
@@ -102,25 +100,29 @@ export class ImportGraph {
...
-    sf.statements.forEach(stmt => {
+    // Look through the source file for import and export statements.
+    for (const stmt of sf.statements) {
       if ((ts.isImportDeclaration(stmt) || ts.isExportDeclaration(stmt)) &&
-          stmt.moduleSpecifier !== undefined && ts.isStringLiteral(stmt.moduleSpecifier)) {
-        // Resolve the module to a file, and check whether that file is in the ts.Program.
-        const moduleName = stmt.moduleSpecifier.text;
-        const moduleFile = this.resolver.resolveModule(moduleName, sf.fileName);
-        if (moduleFile !== null && isLocalFile(moduleFile)) {
+          stmt.moduleSpecifier !== undefined) {
```
Comment (alxhub) [inline packages/compiler-cli/src/ngtsc/cycles/src/imports.ts:106]: A stylistic note: let's restructure this a bit and fix the poor structure left by the original author (who shall remain nameless and is certainly not me...). We can invert the logic so we check each condition as we go and `continue` if it doesn't meet our expectations. Then the `imports.add` final effect can be at the top level of the loop if we successfully pass all preconditions.

**Review by alxhub — APPROVED** [review]

File: packages/compiler-cli/src/ngtsc/cycles/src/imports.ts (hunk 1 above)
Comment (JoostK) [reply packages/compiler-cli/src/ngtsc/cycles/src/imports.ts:106]: Haha, I actually did switch it around and then went back to keep the diff smaller :-) I'll change it to use `continue`.

**Comment by JoostK** [comment]
Marking this as blocked on #40947 because I know they conflict

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:361592737:BASE:361592776:1615226457488:8c8eba7f).

**Pull request closed without merging**